base/sequence_clock.go

```diff
@@ -291,6 +292,23 @@ func (c *SequenceClockImpl) UpdateWithClock(updateClock SequenceClock) {
 	}
 }
 
+func (c *SequenceClockImpl) UpdateWithPartitionClocks(partitionClocks []*PartitionClock, allowRollback bool) {
```

Comment:
Since this is updating the *SequenceClockImpl, can you add a comment saying that a lock must be held in order to be goroutine safe? (I wish there was a better way than this ..)